samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -210,7 +198,9 @@ class TaskInstance(
 
     trace("Flushing producers for taskName: %s" format taskName)
 
-    collector.flush
+    exceptionHandler.maybeHandle { // Allow users to decide whether to swallow producer exceptions (in kafka whole batch will be lost!)
+      collector.flush
```

Comment:
Good question. 

If they want to ignore producer exceptions (and understand that they will drop whole batches of output messages if they do), they just add the config:
task.ignored.exceptions=org.apache.samza.system.SystemProducerException

See the following unit test for an example:
org.apache.samza.container.TestTaskInstance#testProducerExceptionsCanBeIgnored